### PR TITLE
[CARBONDATA-1649]insert overwrite fix during job interruption

### PR DESCRIPTION
**Pull request by akashrn5**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [*] Make sure the PR title is formatted like:
   `[CARBONDATA-1649>] 
    when insert overwrite job is in progress and if that job is cancelled, then table status should be 
    changed and segment should be deleted during the interruption
   
 - [*] Make sure to add PR description including
       Root cause - when insert overwrite job is in progress and if that job is cancelled, then table status 
       was not getting updated and after that insert or any load was not possible 
       Implemented Solution- when the insert overwrite is interrupted, then update the table status to 
      failure
 - [*] Any interfaces changed?
       NONE

 - [*] Any backward compatibility impacted?
      NONE

 - [*] Document update required?
     NONE

 - [*] Testing done
     Tested and there is no performance hit for this
      
         
 - [*] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
             NONE    
---

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/686/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/690/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1323/

**Comment by akashrn5** [comment]
@ravipesala @gvramana @QiangCai @manishgupta88 please review this PR

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1119/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1717/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/LoadTableCommand.scala [hunk 1]
```diff
@@ -186,6 +186,12 @@ case class LoadTableCommand(
           LOGGER.error(ex, s"Dataload failure for $dbName.$tableName")
           throw new RuntimeException(s"Dataload failure for $dbName.$tableName, ${ex.getMessage}")
         case ex: Exception =>
+          if (ex.isInstanceOf[InterruptedException] &&
+              ex.getMessage.contains("update fail status")) {
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/LoadTableCommand.scala]: It is not good to rely on message inside the exception, can you create one special exception for this case?

Comment (akashrn5) [reply integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/LoadTableCommand.scala]: ok, i will add a new Exception for this, and throw that exception

**Comment by akashrn5** [comment]
@jackylk handled your comment, please review

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1167/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1745/

**Comment by jackylk** [comment]
Please rebase @akashrn5

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1411/

**Pull request closed without merging**